lock-impl/src/main/java/com/palantir/lock/impl/LockRefreshingTimelockService.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.impl;
+
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.palantir.lock.v2.LockImmutableTimestampRequest;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequestV2;
+import com.palantir.lock.v2.LockTokenV2;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.v2.WaitForLocksRequest;
+import com.palantir.timestamp.TimestampRange;
+
+// TODO(nziebart): probably should make it more obvious that this class should always be used;
+// maybe call this a TimelockClient and require that everywhere? Could also be used for async unlocking..
+public class LockRefreshingTimelockService implements TimelockService {
```

Comment:
Discussed briefly in person. The idea here is that we want to keep locks alive as long as the server is running; this allows us to set the lease expiration time to a lower number, which is better if a server dies while holding locks.

Also relevant is that we currently wrap `RemoteLockService` in `LockRefreshingLockService` which does the same thing.. so I basically followed the same pattern here as I didn't see an obvious reason to change that behavior.